flake.nix

```diff
@@ -180,6 +180,19 @@
         ghc8104 = pkgs.hlsHpkgs "ghc8104";
         ghc901 = ghc901Config.tweakHpkgs (pkgs.hlsHpkgs "ghc901");
 
+        # For markdown support
+        myst-parser = pkgs.python3Packages.callPackage ./myst-parser.nix {};
+        sphinx_rtd_theme = pkgs.python3Packages.sphinx_rtd_theme.overrideAttrs (oldAttrs: {
+          # For https://github.com/readthedocs/sphinx_rtd_theme/pull/1185, otherwise lists are broken locally
+          src = pkgs.fetchFromGitHub {
+            owner = "readthedocs";
+            repo = "sphinx_rtd_theme";
+            rev = "34f81daaf52466366c80003db293d50075c1b896";
+            sha256 = "0rkrsvvqr6g2p3v5vq88jhfp5sd0r1jqjh3vc5y26jn30z8s4fkz";
+          };
+        });
+        pythonWithPackages = pkgs.python3.withPackages (ps: [ps.sphinx myst-parser sphinx_rtd_theme]);
+
```

Comment:
Can we wrap them into a derivation, which will essentially invoke `docs/Makefile` ?

Reply:
We can definitely have a derivation that builds the doc site, but it's nice to have sphinx and the required packages in the shell so you can build them easily when you're working on them.

Reply:
> but it's nice to have sphinx and the required packages in the shell so you can build them easily when you're working on them

Yes, but we can enter the dev shell of that derivation, not HLS's. So people don't need to download and build sphinx things while developing the Haskell parts.

Reply:
Personally I find it nice if running `nix-shell` gives you what you need for everything in the project. Otherwise it can be quite obscure to figure out how to get the right shell. As it is, there is a "blessed" shell which is `devShell`/what you get if you just run `nix-shell`. I think the convenience of having that "just work" outweighs not having to download Sphinx.

Reply:
I pushed a derivation for the docs.

Reply:
i know I would ask for a chimera, dear fellow *nix hackers, but it would be nice to be able to build the website in windows native...
will investigate myself 😝